src/Circle.js

```diff
@@ -0,0 +1,95 @@
+import React, { useRef, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const DIRECTIONS = {
+    CLOCKWISE: 'clockwise',
+    ANTI_CLOCKWISE: 'antiClockwise',
+    BOTH_SIDES: 'bothSides',
+};
+
+const getSvgClassName = (strokePercentage, direction, className) => classNames(
+    'react-circle_svg',
+    `react-circle_${direction}`,
+    {
+        'react-circle_noStroke': strokePercentage === 0,
+    },
+    className,
+);
+
+const Circle = ({ className, direction, strokeWidth, strokePercentage, onTransitionEnd }) => {
+    const svgRef = useRef(null);
+    const [circleState, setCircleState] = useState({});
+    const [svgState, setSvgState] = useState({ className: getSvgClassName(strokePercentage, direction, className) });
+
+    useEffect(() => {
+        const { clientWidth: svgSize } = svgRef.current;
+
+        const center = svgSize / 2;
+        const radius = center - parseInt(strokeWidth, 10);
+        const perimeter = 2 * Math.PI * radius;
+        const rotate = (-strokePercentage * 180);
+        const strokeLineSize = perimeter * strokePercentage;
+        const strokeGapSize = perimeter - strokeLineSize;
+
+        const strokeDasharray = `${strokeLineSize} ${strokeGapSize}`;
+
+        const circleStyle = {
+            strokeWidth,
+            strokeDasharray,
+        };
+        const groupStyle = {};
+
+        if (direction === DIRECTIONS.BOTH_SIDES) {
+            groupStyle.transform = `rotate(${rotate}deg)`;
+        }
+
+        setCircleState({
+            center,
+            radius,
+            perimeter,
+            rotate,
+            strokeGapSize,
+            strokeLineSize,
+            strokeDasharray,
+            circleStyle,
+            groupStyle,
+        });
+
+        setSvgState({ className: getSvgClassName(strokePercentage, direction, className) });
+    }, [svgRef, strokeWidth, strokePercentage, direction, className]);
+
+    return (
+        <svg
+            ref={ svgRef }
+            className={ svgState.className }>
+            { svgRef.current &&
+                <g style={ circleState.groupStyle }>
+                    <circle
+                        style={ circleState.circleStyle }
+                        cx={ circleState.center }
+                        cy={ circleState.center }
+                        r={ circleState.radius }
+                        onTransitionEnd={ onTransitionEnd } />
+                </g>
+            }
+        </svg>
+    );
+};
+
+Circle.propTypes = {
+    className: PropTypes.string,
+    direction: PropTypes.oneOf(Object.values(DIRECTIONS)),
+    strokeWidth: PropTypes.number,
+    strokePercentage: PropTypes.number,
+    onTransitionEnd: PropTypes.func,
+};
+
+Circle.defaultProps = {
+    direction: DIRECTIONS.CLOCKWISE,
+    strokeWidth: 1.5,
+    strokePercentage: 1,
+    onTransitionEnd: /* istanbul ignore next */ () => {},
```

Comment:
Don't you think it's a little bit weird to have a default `onTransitionEnd` callback that does nothing?

Reply:
This was something that was present in the extracted component, I thought it was because when you pass `undefined` to the `onTransitionEnd` property of the `circle` element, an error of `undefined is not a function` would result.

I tried it out and it seems like that is not the case, so I'll probably remove it. @paulobmarcos Is there any reason for this that I'm not seeing?

Reply:
Its a `noop`. At the time it felt strange leaving it as undefined. But you can remove it if you want.